frontend/public/module/k8s/k8s-models.ts

```diff
@@ -27,7 +27,7 @@ export const modelsToMap = (models: K8sKind[]): ImmutableMap<K8sResourceKindRefe
 let k8sModels = modelsToMap(_.values(staticModels));
 
 const hasModel = (model: K8sKind) =>
```

Comment:
Suggested change:

```
const modelKey = (model) => {
  // TODO: Use `referenceForModel` even for known API objects
  return model.crd ? referenceForModel(model) : model.kind;
};

export const modelsToMap = (models: K8sKind[]): ImmutableMap<K8sResourceKindReference, K8sKind> => {
  return ImmutableMap<K8sResourceKindReference, K8sKind>().withMutations((map) => {
    models.forEach((model) => map.set(modelKey(model), model));
  });
};
...
const hasModel = (model: K8sKind) => k8sModels.has(modelKey(model));
```

That way they are both based off the same value, it looks it up in the same fashion it stores it. No way for it to be confused on reverse lookups.